src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -1151,47 +1151,68 @@ module Types = struct
           ; abstract_field "toAccount"
               ~typ:(non_null AccountObj.account)
               ~args:[] ~doc:"Account of the receiver"
-              ~deprecated:(Deprecated (Some "use receiver field instead")) ] )
+              ~deprecated:(Deprecated (Some "use receiver field instead"))
+          ; abstract_field "failureReason" ~typ:string ~args:[]
+              ~doc:"null is no failure, reason for failure otherwise." ] )
+
+    module Status = struct
+      type t =
+        | Applied
+        | Included_but_failed of Transaction_status.Failure.t
+        | Unknown
+    end
+
+    module With_status = struct
+      type 'a t = {data: 'a; status: Status.t}
+
+      let map t ~f = {t with data= f t.data}
+    end
+
+    let field' ?doc ?deprecated lab ~typ ~args ~resolve =
```

Comment:
would this be better-named as something like `field_with_status_data`

Reply:
I like `field'` actually since it's used so often